src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/DeleteTeiAction/DeleteTeiAction.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import React, { useState } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { Button, ButtonStrip, Modal, ModalActions, ModalContent, ModalTitle } from '@dhis2/ui';
+import { useAuthority } from '../../../../../../utils/userInfo/useAuthority';
+import { useCascadeDeleteTei } from './hooks/useCascadeDeleteTei';
+
+type Props = {
+    selectedRows: { [id: string]: boolean },
+    selectedRowsCount: number,
+    trackedEntityName: string,
+    onUpdateList: () => void,
+}
+
+const CASCADE_DELETE_TEI_AUTHORITY = 'F_TEI_CASCADE_DELETE';
+
+export const DeleteTeiAction = ({
+    selectedRows,
+    selectedRowsCount,
+    trackedEntityName,
+    onUpdateList,
+}: Props) => {
+    const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+    const { hasAuthority } = useAuthority({ authority: CASCADE_DELETE_TEI_AUTHORITY });
+    const { deleteTeis, isLoading } = useCascadeDeleteTei({
+        selectedRows,
+        setIsDeleteDialogOpen,
+        onUpdateList,
+    });
+
+    if (!hasAuthority) {
+        return null;
+    }
+
+    return (
+        <>
+            <Button
```

Comment:
The DeleteTeiaction action is missing the program access and trackedEntityType checks. Can something similar to how [WidgetProfile](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js#L122-L124) currently works be implemented? Thanks!